### PR TITLE
BREAKING CHANGE: New UI for messages on the pci-sdk-web (1201415348104520)

### PR DESCRIPTION
**Pull request by Claratj**
I'm going to update the docs to explain this breaking change.

### CODE REVIEW

File: packages/pci-sdk-iframe/src/apiClient/errorMessage.parser.ts [hunk 1]
```diff
@@ -3,11 +3,11 @@ function parse401(errorCode: number, serverMessage: string): string {
 		case 90263:
 			return 'Cardholder needs to verify their identity. Sending 2FA code to phone or email...';
 		case 3032:
-			return 'The user token you provided is invalid. Please set a valid user token: https://docs.aptopayments.com/docs/sdks/Web/pci_sdk_web/#optionsobject-properties';
+			return 'Invalid user token. Check the PCI SDK Web docs for more info.';
 		case 3035:
-			return 'The mobile API key you provided is invalid. Please set a valid API key: https://docs.aptopayments.com/docs/sdks/Web/pci_sdk_web/#get-the-mobile-api-key';
+			return 'Invalid Mobile API key. Check the PCI SDK Web docs for more info.';
```
Comment (0ubbe) [inline packages/pci-sdk-iframe/src/apiClient/errorMessage.parser.ts]: This could be user-facing, right? So I would instead rephrase it to:
```
Invalid Mobile API key. Contact support.
```
As this is displayed to the users in our clients' application, the user will not know where are the docs for **PCI SDK Web** are and maybe grow frustrated?

File: packages/pci-sdk-iframe/src/pages/app/components/Notification/Notification.css [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+@keyframes slide-up {
+	from {
+		bottom: -100%;
+	}
+	to {
+		bottom: 0;
+	}
+}
```
Comment (0ubbe) [inline packages/pci-sdk-iframe/src/pages/app/components/Notification/Notification.css]: For [performance](https://stackoverflow.com/questions/13410723/transformtranslatex-vs-transition-on-left-property-which-has-better-performanc), we might want to do this with `transform: translateY(-100%)` instead.

IIRC `transform` does better on mobile devices and it seems we're moving the element quite a lot here.

File: packages/pci-sdk-iframe/src/pages/app/components/Notification/Notification.test.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		it('should render a positive notification when type is negative', () => {
+			render(<Notification message="dummy_notification" type="negative" />);
+
+			expect(screen.queryByTestId('notification')).toHaveClass('Notification--negative');
+		});
+
+		it('should style the notification when type is positive and a custom theme is being passed', () => {
+			render(<Notification message="dummy_notification" type="positive" theme={dummy_style_full_custom_positive} />);
+
+			expect((screen.queryByTestId('notification') as HTMLDivElement).getAttribute('style')).toBe(
```
Comment (0ubbe) [inline packages/pci-sdk-iframe/src/pages/app/components/Notification/Notification.test.tsx]:
```suggestion
			expect((screen.getByTestId('notification')).getAttribute('style')).toBe(
```

File: packages/pci-sdk-iframe/src/pages/app/components/Notification/Notification.test.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+				'background-color: green;'
+			);
+		});
+
+		it('should style the notification when type is negative and a custom theme is being passed', () => {
+			render(<Notification message="dummy_notification" type="negative" theme={dummy_style_full_custom_negative} />);
+
+			expect((screen.queryByTestId('notification') as HTMLDivElement).getAttribute('style')).toBe(
```
Comment (0ubbe) [inline packages/pci-sdk-iframe/src/pages/app/components/Notification/Notification.test.tsx]:
```suggestion
			expect((screen.getByTestId('notification')).getAttribute('style')).toBe(
```

File: packages/pci-sdk-iframe/src/services/cardData.service.ts [hunk 5]
```diff
@@ -64,7 +64,12 @@ function showCardData({ dispatch, cardId, isPCICompliant }: IShowCardDataArgs) {
 		.then((card) => dispatch({ ...card, isLoading: false, message: '', uiStatus: 'CARD_DATA_VISIBLE' }))
 		.catch((err) => {
 			if (_checkIfInvalidAPIKeyError(err)) {
-				return dispatch({ message: 'Invalid API key', uiStatus: 'CARD_DATA_HIDDEN', isLoading: false });
+				return dispatch({
+					message: 'Invalid mobile API key. Refer to the Developer Portal for a valid key.',
+					uiStatus: 'CARD_DATA_HIDDEN',
+					isLoading: false,
+					notificationType: 'negative',
+				});
```
Comment (0ubbe) [inline packages/pci-sdk-iframe/src/services/cardData.service.ts:72]: Should we open a ticket in **Asana** to write tests verifying these new error messages are shown correctly?

**Review by 0ubbe — COMMENTED** [review]
Looking good 💚 , nothing blocking from my side, some small questions 💬

File: packages/pci-sdk-iframe/src/apiClient/errorMessage.parser.ts (hunk 1 above)
Comment (IagoLast) [reply packages/pci-sdk-iframe/src/apiClient/errorMessage.parser.ts]: 🧠

File: packages/pci-sdk-iframe/src/apiClient/index.ts [hunk 6]
```diff
@@ -104,11 +104,7 @@ export async function getCardData(cardId: string, auth?: { verificationId: strin
 		case 401:
 			throw new Error(errorMessageParser.parse401(data.code, data.message));
 		case 400:
-			throw new Error(
-				errorMessageParser.parse400(
-					'Invalid request. Are you sure the cardID is correct? https://docs.aptopayments.com/docs/sdks/Web/pci_sdk_web/#optionsobject-properties'
-				)
-			);
+			throw new Error(errorMessageParser.parse400('Invalid cardID. Check the PCI SDK Web docs for more info.'));
```
Comment (IagoLast) [inline packages/pci-sdk-iframe/src/apiClient/index.ts:107]: I think the link was very useful 😕

File: packages/pci-sdk-iframe/src/apiClient/index.ts [hunk 7]
```diff
@@ -157,11 +153,7 @@ async function setPin(args: ISetPinArgs) {
...
```
Comment (IagoLast) [inline packages/pci-sdk-iframe/src/apiClient/index.ts:156]: Same, the link was very useful 😕

**Review by IagoLast — COMMENTED** [review]
I agree on lluis coments also:

- I think we should create a type to represent Notification types
- I liked the links on the errors

**Review by 0ubbe — APPROVED** [review]
